src/rocksdb_options.rs

```diff
@@ -311,27 +311,105 @@ unsafe impl Send for Statistics {}
 unsafe impl Sync for Statistics {}
 
 impl Statistics {
-    pub fn new() -> Self {
+    pub fn new() -> Arc<Self> {
```

Comment:
Why need `Arc`? Note `statistics` is already a shared_ptr. I think it will be simpler to recommend user to store statistics somewhere instead of fetching from `option`.

Reply:
It's a better mental model than passing a reference around. After this PR there's no way to create two `Statistics` objects that maps to the same shared_ptr. So the rust side structure is consistent with C++ side.

Reply:
> If we don't allow fetching statistics from options

This is too inconvenient. Option A, we disallow cloning the statistics, then the user must manually maintain a global copy of statistics if he ever wants to use it after the DB creation. Option B, we allow cloning the statistics, then it would just be a bad mental model and the allocation cost is bitter than modifying the reference counter.

Reply:
Option A is the best way I think. Maintaining a global copy is not that hard. If user wants, they can also wrap the statistics into `Arc` outside of the library, then clone and pass it around. Just pay as they use.